internal/provider/providers/porkbun/provider.go

```diff
@@ -138,8 +138,22 @@ func (p *Provider) Update(ctx context.Context, client *http.Client, ip netip.Add
 		return ip, nil
 	}
 
-	for _, recordID := range recordIDs {
-		err = p.updateRecord(ctx, client, recordType, ipStr, recordID)
+	for _, record := range records {
+		matches := record.Name == (p.owner+"."+p.domain) ||
+			(p.owner == "@" && record.Name == p.domain)
+
+		if !matches {
+			// We fetch by domain, type, and owner. This should never happen
+			return netip.Addr{}, fmt.Errorf("received unexpected record for non matching owner: %w", errors.ErrUnknownResponse)
+		}
+
+		if record.Content == ipStr {
+			// Skip update. Porkbun already has the IP correctly set.
```

Comment:
1. Did you face this problem in practice?
2. AFAIK this can only happen if the dns server is caching the old records due to a large TTL, although the update has been sent. In that case, we should address this at "upper" layers in the code so it applies to all providers. For example, if an update was sent, even if the resolved IP address mismatches your public IP address, do not attempt to send another update for the TTL value (optional, defaults to 300 seconds, configurable in the JSON of each record). I would like to suggest to revert the changes related to this here (i.e. `getRecords` back to `getRecordIDs` etc. and address this in another PR.

Reply:
1. So, I did face it. I would remove the updates.json, and then get into a bad state where it would attempt to update the entry, but then get a 400 error.
I've done some more testing and I can't repro it, and I think it must have been all related to the wildcard issue, because with the wild card issue, I would end up with multiple DNS entries, one for 127.0.0.1, one for my actual IP. I believe Porkbun under the hood stores the *.domain.tld differently from the %2A.domain.tld hence update would attempt to update the %2A.domain.tld, but it would be correct already and then fail.

It's probably not necessary at this time.

Reply:
The primary issue that this does solve is the issue of, when you're first setting up ddns-updater, if the DNS entry hasn't propogated fully and you remove your updates.json, for example, because something in it gets messed up, then subsequent runs of ddns-updater will attempt to update the dns values, get 400, and then fail out from then on because it keeps trying to update the DNS entry. So basically, once you get into the bad 400 state, it will never fix itself, because it will indefinitely try to update the DNS entry, but the DNS entry is actually correct.

Additionally in v2.7.0, my first entry had just my single domain.tld, and if I didn't put in `"host":"@"` it wouldn't work. that doesn't really seem to be the case on the current master though.

So the confluence of the 3 bugs:
400 error when attempting to edit a domain that hasn't fully propogated to the DNS records
*.domain.tld being broken causing it to get into a state it could never resolve (constantly trying to update the same correct entry, 400 error, etc...)
the `@` host not being documented but being required for 2.7.0 (or it seemed like it was), causing me to get into a state where I removed my updates.json, then causing more 400 errors, because the entries were correct but erroring when it tried to update

All resulted in me almost giving up on using DNS-Updater, and I have seen other comments on reddit to that affect, and I can only assume they ran into the same problems I did. 


I'll split everything out into separate PRs, and we can discuss / merge or trash whatever. They're already all split up, I'll just divide and PR individually.

Reply:
sorry, didn't read what you put fully for 2.
so yes, this is primarily a fix for (2) as you stated, and yes, fixing at a higher level is probably a good idea.

I do feel like the context of why each of these fixes came about as without them, I got into a quite bad state very early in my setup. heh

Reply:
I split into #774 which we can probably just close as won't merge and update as you say at a higher level in the logic.